backend/src/services/project/project-dal.ts

```diff
@@ -30,8 +30,33 @@ export const projectDALFactory = (db: TDbClient) => {
           { column: `${TableName.Environment}.position`, order: "asc" }
         ]);
 
+      const groups: string[] = await db(TableName.UserGroupMembership)
+        .where({ userId })
+        .select(selectAllTableCols(TableName.UserGroupMembership))
+        .pluck("groupId");
+
+      const groupWorkspaces = await db(TableName.GroupProjectMembership)
```

Comment:
This could be combined into one singular query. I wouldn't consider it a blocker for the PR, but feel free to look into it if you want, or just resolve if you won't look into it for now.

Reply:
@akhilmhdh mentioned this as well but I think for the purposes of this PR this should be ok — We can optimize it in the future.